pyteal/ast/itxn_test.py

```diff
@@ -459,240 +459,50 @@ def test_InnerTxnBuilder_method_call(
-            pt.InnerTxnBuilder.Next(),
-            pt.InnerTxnBuilder.SetFields(t5_3),  # type: ignore
-            pt.InnerTxnBuilder.Next(),
-            pt.InnerTxnBuilder.SetFields(
-                {
-                    pt.TxnField.type_enum: TxnType.ApplicationCall,
-                    pt.TxnField.application_id: pt.Int(1),
-                    pt.TxnField.application_args: [
-                        pt.MethodSignature("add(pay,txn,appl)void"),
-                    ],
-                    pt.TxnField.fee: pt.Int(0),
-                }
-            ),
-        ),
-        None,
-    ),
-    # Error cases
-    (
-        pt.Int(1),
-        "add(pay,txn,appl)void",
-        [
-            {},
-            {TxnField.type_enum: TxnType.AssetTransfer},
-            {TxnField.type_enum: TxnType.ApplicationCall},
-        ],
-        None,
-        None,
-        pt.TealInputError,
-    ),
-    (
-        pt.Int(1),
-        "add(pay,txn,appl)void",
-        [
-            {TxnField.type_enum: pt.Int(10)},
-            {TxnField.type_enum: TxnType.AssetTransfer},
-            {TxnField.type_enum: TxnType.ApplicationCall},
-        ],
-        None,
-        None,
-        pt.TealTypeError,
-    ),
-    (
-        pt.Int(1),
-        "add(pay,txn,appl)void",
-        [
-            {TxnField.type_enum: TxnType.ApplicationCall},
-            {TxnField.type_enum: TxnType.AssetTransfer},
-            {TxnField.type_enum: TxnType.ApplicationCall},
-        ],
-        None,
-        None,
-        pt.TealInputError,
-    ),
-    (
-        pt.Int(1),
-        "add(application,account,asset)void",
-        [
-            pt.abi.Asset(),
-            pt.abi.Account(),
-            pt.abi.Asset(),
-        ],
-        None,
-        None,
-        pt.TealTypeError,
-    ),
-    (
-        pt.Int(1),
-        "add(application)void",
-        [
-            pt.Bytes(""),
-        ],
-        None,
-        None,
-        pt.TealTypeError,
-    ),
-    (
-        pt.Int(1),
-        "add(asset)void",
-        [
-            pt.Bytes(""),
-        ],
-        None,
-        None,
-        pt.TealTypeError,
-    ),
-    (
-        pt.Int(1),
-        "add(account)void",
-        [
-            pt.Int(1),
-        ],
-        None,
-        None,
-        pt.TealTypeError,
-    ),
-    (
-        pt.Int(1),
-        "add(uint64,uint64)void",
-        [pt.abi.String(), pt.abi.Uint64()],
-        None,
-        None,
-        pt.TealTypeError,
-    ),
-    (
-        pt.Int(1),
-        "add(uint64,uint64)void",
-        [pt.abi.Uint64()],
-        None,
-        None,
-        pt.TealInputError,
-    ),
-    (
-        pt.Int(1),
-        "add(uint64,uint64)void",
-        [pt.abi.Uint64(), pt.abi.Uint64(), pt.abi.Uint64()],
-        None,
-        None,
-        pt.TealInputError,
-    ),
+@pytest.mark.parametrize(
+    "app_id, sig, args, extra_fields, expected_expr, expected_error", ITXN_METHOD_CASES
 )
+def test_InnerTxnBuilder_execute_method(
```

Comment:
I think this test is largely redundant with `test_InnerTxnBuilder_method_call`, are they mergable?

Reply:
I combined them 👍